neqo-transport/src/send_stream.rs

```diff
@@ -696,6 +697,67 @@ impl SendStream {
     }
 }
 
+#[derive(Debug, Default)]
+pub(crate) struct SendStreams(BTreeMap<StreamId, SendStream>);
+
+impl SendStreams {
+    pub(crate) fn get(&self, id: StreamId) -> Res<&SendStream> {
+        self.0.get(&id).ok_or_else(|| Error::InvalidStreamId)
+    }
+
+    pub(crate) fn get_mut(&mut self, id: StreamId) -> Res<&mut SendStream> {
+        self.0.get_mut(&id).ok_or_else(|| Error::InvalidStreamId)
+    }
+
+    pub(crate) fn insert(&mut self, id: StreamId, stream: SendStream) {
+        self.0.insert(id, stream);
+    }
+
+    pub(crate) fn acked(&mut self, token: StreamGeneratorToken) {
+        if let Some(ss) = self.0.get_mut(&token.id) {
+            ss.mark_as_acked(token.offset, token.length as usize, token.fin);
+        }
+    }
+
+    pub(crate) fn reset_acked(&mut self, id: StreamId) {
+        if let Some(ss) = self.0.get_mut(&id) {
+            ss.reset_acked()
+        }
+    }
+
+    pub(crate) fn lost(&mut self, token: StreamGeneratorToken) {
+        if let Some(ss) = self.0.get_mut(&token.id) {
+            ss.mark_as_lost(token.offset, token.length as usize, token.fin);
+        }
+    }
+
+    pub(crate) fn clear(&mut self) {
+        self.0.clear()
+    }
+
+    pub(crate) fn clear_terminal(&mut self) {
+        let send_to_remove: SmallVec<[_; 8]> = self
+            .0
+            .iter()
+            .filter(|(_, stream)| stream.is_terminal())
+            .map(|(id, _)| *id)
+            .collect();
+
+        for id in send_to_remove {
+            self.0.remove(&id);
+        }
+    }
+}
+
+impl<'a> IntoIterator for &'a mut SendStreams {
```

Comment:
Is this something you need?

Reply:
frame generator mutable iterates streams.